src/Altinn.App.Core/Features/Validation/Default/SigningTaskValidator.cs

```diff
@@ -0,0 +1,159 @@
+using Altinn.App.Core.Helpers.Serialization;
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Internal.Data;
+using Altinn.App.Core.Internal.Instances;
+using Altinn.App.Core.Internal.Process;
+using Altinn.App.Core.Internal.Process.Elements.AltinnExtensionProperties;
+using Altinn.App.Core.Models;
+using Altinn.App.Core.Models.Validation;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.App.Core.Features.Validation.Default;
+
+/// <summary>
+/// Default validator for signing tasks. Validates that all parties have signed the current task.
+/// </summary>
+internal sealed class SigningTaskValidator : IValidator
+{
+    private readonly IProcessReader _processReader;
+    private readonly ISigningService _signingService;
+    private readonly IDataClient _dataClient;
+    private readonly IInstanceClient _instanceClient;
+    private readonly IAppMetadata _appMetadata;
+    private readonly ModelSerializationService _modelSerialization;
+    private readonly ILogger<SigningTaskValidator> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SigningTaskValidator"/> class.
+    /// </summary>
+    public SigningTaskValidator(
+        ILogger<SigningTaskValidator> logger,
+        IProcessReader processReader,
+        ISigningService signingService,
+        IDataClient dataClient,
+        IInstanceClient instanceClient,
+        IAppMetadata appMetadata,
+        ModelSerializationService modelSerialization
+    )
+    {
+        _logger = logger;
+        _processReader = processReader;
+        _signingService = signingService;
+        _dataClient = dataClient;
+        _instanceClient = instanceClient;
+        _appMetadata = appMetadata;
+        _modelSerialization = modelSerialization;
+    }
+
+    /// <summary>
+    /// We implement <see cref="ShouldRunForTask"/> instead.
+    /// </summary>
+    public string TaskId => "*";
+
+    /// <summary>
+    /// Only runs for tasks that are of type "signing".
+    /// </summary>
+    public bool ShouldRunForTask(string taskId)
+    {
+        AltinnTaskExtension? taskConfig;
+        try
+        {
+            taskConfig = _processReader.GetAltinnTaskExtension(taskId);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, $"Error while fetching signing configuration for task {taskId}");
+            return false;
+        }
+
+        AltinnSignatureConfiguration? signingConfiguration = taskConfig?.SignatureConfiguration;
+
+        return signingConfiguration?.RunDefaultValidator == true && taskConfig?.TaskType is "signing";
+    }
+
+    public bool NoIncrementalValidation => true;
+
+    /// <summary>
+    /// Should never be called because NoIncrementalValidation is true.
+    /// </summary>
+    public Task<bool> HasRelevantChanges(IInstanceDataAccessor dataAccessor, string taskId, DataElementChanges changes)
+    {
+        throw new NotImplementedException();
```

Comment:
We can add the documentation as a message, so that it is easier to understand why this is not implemented.
```suggestion
        throw new UnreachableException("HasRelevantChanges should never be called because NoIncrementalValidation is true.");
```
This might need to import `using System.Diagnostics;`